### PR TITLE
Swift 4 support

### PR DESCRIPTION
**Pull request by AYastrebov**
- Convert to swift 4
- Remove warnings

### CODE REVIEW

File: Sources/Queries/SequenceQuery.swift
Comment (hectormatos2011) [inline Sources/Queries/SequenceQuery.swift:52]: Any reason for these removals?

File: Sources/Protocols/Reflectable.swift
Comment (hectormatos2011) [inline Sources/Protocols/Reflectable.swift]: I'm typically not a fan of including implicitly unwrapped optionals. Could you pretty please remove it? You could do this as well:

```swift
if let currentProperty = properties?[i], let propertyName = String(validatingUTF8: property_getName(currentProperty))
```

File: Sources/Extensions/NSPredicate+PrediKit.swift
Comment (hectormatos2011) [inline Sources/Extensions/NSPredicate+PrediKit.swift:22]: Why are we removing `Reflectable` here? What was the problem it caused? If I remember correctly, I   put this here to enforce the developer to ensure that whatever type they feed into this conforms to `Reflectable`. The only class conforming to it is `NSObject` at the moment since this library was intended for use with CoreData. Is there a case where you can use PrediKit with straight up Swift-only objects? Is there a particular API where that's needed?

File: Sources/Queries/MemberQuery.swift
Comment (hectormatos2011) [inline Sources/Queries/MemberQuery.swift:14]: See the above question about removing Reflectable. The same case goes for the AnyObject constraints I put here.

**Review by hectormatos2011 — CHANGES_REQUESTED** [review]
Thank you so much for putting time into updating this! I would absolutely love to support Swift 4 in PrediKit. So, you're awesome <3

I do, however, have a couple of questions and need a couple of things changed or questions addressed before merging this.

At the moment, it looks like this PR doesn't pass tests. Could you look into which tests are failing and fix them? I generally won't approve a PR if the tests aren't passing.

**Comment by AYastrebov** [comment]
@hectormatos2011 tests are passing now. I've updated travis config also

File: Sources/Queries/SequenceQuery.swift
Comment (AYastrebov) [reply Sources/Queries/SequenceQuery.swift:52]: Tha was done to remove the warning ```SequenceQuery.swift:52:174: Redundant conformance constraint 'U': 'Reflectable'```

File: Sources/Extensions/NSPredicate+PrediKit.swift
Comment (AYastrebov) [reply Sources/Extensions/NSPredicate+PrediKit.swift:22]: The same was here ```NSPredicate+PrediKit.swift:22:25: Redundant conformance constraint 'T': 'Reflectable'```

File: Sources/Queries/MemberQuery.swift
Comment (AYastrebov) [reply Sources/Queries/MemberQuery.swift:14]: And here ```MemberQuery.swift:14:74: Redundant layout constraint 'MemberType' : 'AnyObject'```

File: Sources/Extensions/NSPredicate+PrediKit.swift
Comment (hectormatos2011) [reply Sources/Extensions/NSPredicate+PrediKit.swift:22]: Good to know! Makes sense now too

**Comment by hectormatos2011** [comment]
Again, thank you! If I remember someone today I'll make sure to update CocoaPods to reflect the new version

**Pull request merged**
